_docs/04-adding-checkers.md

```diff
@@ -89,63 +89,52 @@ let callback_my_simple_checker { Callbacks.proc_desc; proc_name } =
 
 In this particular example, `Sil.Call` has the following information:
 
 ```ocaml
 Sil.Call (
 	list_of_return_values,
-	Sil.Const (Sil.Cfun name_of_function),
+	Sil.Const (Const.Cfun name_of_function),
 	list_of_arguments,
 	location,
 	call_flags
 )
 ```
 
-I hope this looks straight forward. Argument ```call_flags``` holds information about the function, such as whether it is virtual or not. Again, this is specified in the file ```sil.mli```.
+I hope this looks straight forward. Argument ```call_flags``` holds information about the function, such as whether it is virtual or not. Again, this is specified in the file ```sil.rei```.
 
-The Checker we have written so far is able to detect every single function call. Now, we have to detect whether a specific function call is actually calling ```Printstream.println```.
+The Checker we have written so far is able to detect every single function call. Now, we have to detect whether a specific function call is actually calling ```java.io.PrintStream.println```.
 
 Let's try this:
 
 ```ocaml
+(** Simple Checker *)
 let callback_my_simple_checker { Callbacks.proc_desc; proc_name } =
+  let is_println pln = match pln with
+    | Procname.Java pn_java ->
+        PatternMatch.java_proc_name_with_class_method
+          pn_java "java.io.PrintStream" "println"
+    | _ ->
+        false in
   let do_instr node = function
-    | Sil.Call (_, Sil.Const (Sil.Cfun pn), _, loc, _) when Procname.java_is_print_method pn->
+    | Sil.Call (_, Sil.Const (Const.Cfun pn), _, loc, _) when is_println pn->
+        L.err "Sample Log Message %a@." Procname.pp proc_name;
         ST.report_error
           proc_name
           proc_desc
           "CHECKERS_MY_SIMPLE_CHECKER"
-          location
+          loc
           "A description of my simple checker"
     | _ -> () in
   Cfg.Procdesc.iter_instrs do_instr proc_desc
 
 ```
 
-Can you spot the difference? A new restriction was added to our pattern -- ```Procname.java_is_print_method``` will tell whether this specific function call is a ```println``` function.
-
-```Procname``` module implements several utility functions that are useful for this kind of analysis. However, this particular one ```java_is_print_method``` is not implemented yet.
-
-In file ```procname.ml``` add the following function:
-
-```ocaml
-(** [java_is_print_method pn] returns true if [pn] is an output method **)
-let java_is_print_method = function
-	| Java js ->
-		let _, class_name = js.class_name in
-		(js.method_name = "println") && (class_name = "PrintStream")
-	| _ -> false
-```
-
-And in file ```procname.mli``` add the respective signature:
-
-```ocaml
-(** [java_is_print_method pn] returns true if [pn] is an output method **)
-val java_is_print_method : t -> bool
-```
```

Comment:
I removed the paragraphs because I wrote the `is_println` expression in `checkers.ml` (L117-L122).
Maybe, I should move that to `procname.ml` and retain the content?